### PR TITLE
[TS] LPP-47472 > LPS-171296 Use asset vocabulary name instead

### PR DESCRIPTION
**Pull request by TaiPhamLR**
Dear team,
This PR is for ticket https://issues.liferay.com/browse/LPS-171296 that passed my review. Please help us review it.
Thanks a lot.
cc @sontruongces

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/asset/asset-entry-query-processor-custom-user-attributes/src/main/java/com/liferay/asset/entry/query/processor/custom/user/attributes/internal/CustomUserAttributesAssetEntryQueryProcessor.java [hunk 1]
```diff
@@ -116,8 +116,7 @@ private void _addUserAttributes(
 						assetCategory.getVocabularyId());
 
 				if (Objects.equals(
-						customUserAttributeName,
-						assetVocabulary.getTitleCurrentValue())) {
+						customUserAttributeName, assetVocabulary.getName())) {
```
Comment (AliciaGarciaGarcia) [inline modules/apps/asset/asset-entry-query-processor-custom-user-attributes/src/main/java/com/liferay/asset/entry/query/processor/custom/user/attributes/internal/CustomUserAttributesAssetEntryQueryProcessor.java:119]: Hi @TaiPhamLR 

With this change, the https://issues.liferay.com/browse/LPS-130127 is not working and will result on a regression.

Can you take a look so, both tickets work properly?

Thanks!

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

File: modules/apps/asset/asset-entry-query-processor-custom-user-attributes/src/main/java/com/liferay/asset/entry/query/processor/custom/user/attributes/internal/CustomUserAttributesAssetEntryQueryProcessor.java (hunk 1 above)
Comment (sontruongces) [reply modules/apps/asset/asset-entry-query-processor-custom-user-attributes/src/main/java/com/liferay/asset/entry/query/processor/custom/user/attributes/internal/CustomUserAttributesAssetEntryQueryProcessor.java]: Hi @AliciaGarciaGarcia, thanks for your feedback. I tested the change for LPS-130127 and couldn't reproduce the issue. So please just ignore it. Besides, I updated the code for some missing some case. Thanks!

**Comment by austinchiang** [comment]
ci:test:relevant

File: modules/apps/asset/asset-entry-query-processor-custom-user-attributes/src/main/java/com/liferay/asset/entry/query/processor/custom/user/attributes/internal/CustomUserAttributesAssetEntryQueryProcessor.java [hunk 2]
```diff
@@ -116,8 +116,8 @@ private void _addUserAttributes(
 						assetCategory.getVocabularyId());
 
 				if (Objects.equals(
-						customUserAttributeName,
-						assetVocabulary.getTitleCurrentValue())) {
+						StringUtil.toLowerCase(customUserAttributeName),
```
Comment (AliciaGarciaGarcia) [inline modules/apps/asset/asset-entry-query-processor-custom-user-attributes/src/main/java/com/liferay/asset/entry/query/processor/custom/user/attributes/internal/CustomUserAttributesAssetEntryQueryProcessor.java:119]: Why are you doing this

it's works, but, There is a reason why this should be the same?

could you explain me why?

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

File: modules/apps/asset/asset-entry-query-processor-custom-user-attributes/src/main/java/com/liferay/asset/entry/query/processor/custom/user/attributes/internal/CustomUserAttributesAssetEntryQueryProcessor.java (hunk 2 above)
Comment (sontruongces) [reply modules/apps/asset/asset-entry-query-processor-custom-user-attributes/src/main/java/com/liferay/asset/entry/query/processor/custom/user/attributes/internal/CustomUserAttributesAssetEntryQueryProcessor.java:119]: After testing, I found that if the user creates a **Custom Field** with a lowercase name like 'region'. The customUserAttributeName's value is 'region'. While the value of assetVocabulary.getTitleCurrentValue() or assetVocabulary.getName() is 'Region'. So the condition will be false. So I did it to cover this case. Please let me know your thought.

Comment (sontruongces) [reply modules/apps/asset/asset-entry-query-processor-custom-user-attributes/src/main/java/com/liferay/asset/entry/query/processor/custom/user/attributes/internal/CustomUserAttributesAssetEntryQueryProcessor.java:119]: I'm sorry, the value of **customUserAttributeName** is not **Custom Field** name. It is the value of the **CUSTOM USER ATTRIBUTES** field in Asset Publisher configuration (step 10).

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Pull request closed without merging**